src/main/java/mate/academy/repository/MySqlBookDao.java

```diff
@@ -0,0 +1,120 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Cannot create new book, something gone wrong: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM book WHERE id = ?";
+        try (Connection connection = MySqlConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(ID_PARAMETER_INDEX, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = parseToObject(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get Book from database by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM book";
+        List<Book> resultList = new ArrayList<>();
+        try (Connection connection = MySqlConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = parseToObject(resultSet);
+                resultList.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get books from database", e);
+        }
+        return resultList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE book SET title= ?, price= ? WHERE id = ?";
+        try (Connection connection = MySqlConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(TITLE_PARAMETER_INDEX, book.getTitle());
+            statement.setBigDecimal(PRICE_PARAMETER_INDEX, book.getPrice());
+            statement.setLong(UPDATE_ID_PARAMETER_INDEX, book.getId());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Book was not updated: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM book WHERE id =?";
+        try (Connection connection = MySqlConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(ID_PARAMETER_INDEX, id);
+            return statement.executeUpdate() > ZERO_AFFECTED_ROW;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete book by id " + id, e);
+        }
+    }
+
+    private Book parseToObject(ResultSet resultSet) throws SQLException {
+        long id = resultSet.getObject(ID_ROW_LABEL, Long.class);
```

Comment:
If you are expecting that null can appear then NPE can be thrown here.
```suggestion
        Long id = resultSet.getObject(ID_ROW_LABEL, Long.class);
```